pkg/aspect/clean/clean.go

```diff
@@ -165,3 +189,461 @@ func (c *Clean) Run(_ *cobra.Command, _ []string) error {
+		if readDirErr != nil {
+			// install and cache folders will end up here. Dont want to remove these
+			continue
+		}
+
+		// We expect these folders to have up to 2 files / folders
+		//   - Firstly, a file named "DO_NOT_BUILD_HERE"
+		//   - Secondly, a folder named after the given workspace
+		// We can use the given second folder to determine the name of the workspace. We want this
+		// so that we can ask the user if they want to remove a given workspace
+		if (len(execrootFiles) == 1 && execrootFiles[0].Name() == "DO_NOT_BUILD_HERE") || len(execrootFiles) > 2 {
+			// TODO: Group up unkown workspaces into one prompt?
+			workspaceInfo.workspaceName = "Unknown Workspace"
+		} else {
+			for _, execrootFile := range execrootFiles {
+				if execrootFile.Name() != "DO_NOT_BUILD_HERE" {
+					workspaceInfo.workspaceName = execrootFile.Name()
+				}
+			}
+		}
+
+		sizeCalcQueue <- workspaceInfo
+	}
+}
+
+func (c *Clean) sizeCalculator(
+	in <-chan bazelDirInfo,
+	out chan<- bazelDirInfo,
+	waitGroup *sync.WaitGroup,
+) {
+	for bazelDir := range in {
+		size, hRSize, unit := c.getDirSize(bazelDir.path)
+
+		bazelDir.size = size
+		bazelDir.hRSize = hRSize
+		bazelDir.unit = unit
+		bazelDir.processed = true
+		comparator := bazelDir.accessTime.Hours() * float64(size)
+
+		if bazelDir.isCurrentWorkspace {
+			// If we can avoid cleaning the current working directory then maybe we want to do so?
+			// If the user has selected this mode then they just want to reclaim resources.
+			// Keeping the bazel workspace for the current repo will mean faster build times for that repo.
+			// Dividing by 2 so that the current workspace will be listed later to the user.
+			comparator = comparator / 2
+		}
+
+		bazelDir.comparator = comparator
+
+		out <- bazelDir
+	}
+
+	waitGroup.Done()
+}
+
+func (c *Clean) deleteProcessor(
+	deleteQueue chan bazelDirInfo,
+	waitGroup *sync.WaitGroup,
+	sizeQueue chan<- float64,
+	errors chan<- error,
+) {
+	for bazelDir := range deleteQueue {
+
+		// We know that there will be an "external" folder that could be deleted in parallel.
+		// So we can move those folders to a tmp filepath and add them as seperate deletes that will
+		// therefore happen in parallel.
+		externalFolders, _ := ioutil.ReadDir(filepath.Join(bazelDir.path, "external"))
+		for _, folder := range externalFolders {
+			newPath := c.MoveFolderToTmp(bazelDir.path, folder.Name())
+
+			if newPath != "" {
+				waitGroup.Add(1)
+				deleteQueue <- bazelDirInfo{
+					path: newPath,
+					size: 0, // dont want to calculate the size for the subfolders. Just include the parent size
+				}
+			}
+		}
+
+		// otherwise certain files will throw a permission error when we try to remove them
+		if _, err := c.ChangeFolderPermissions(bazelDir.path); err != nil {
+			waitGroup.Done()
+			errors <- fmt.Errorf("failed to delete %q: %w", bazelDir.path, err)
+			continue
+		}
+
+		// remove entire folder
```

Comment:
```suggestion
		// Remove the entire directory tree.
```

Reply:
I see we have a mix of references to directories versus folders. Could we pick one and use it everywhere?